analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SolutionBuilder.cs

```diff
@@ -78,10 +79,17 @@ internal struct SolutionBuilder
                 throw new ArgumentException("Please use a collection of paths with the same extension", nameof(paths));
             }
 
-            var project = Create()
-                .AddProject(AnalyzerLanguage.FromPath(paths.First()), outputKind: outputKind)
-                .AddDocuments(paths)
-                .AddReferences(additionalReferences);
+            var project = duplicateDocumentsNeeded switch
+            {
+                true => Create()
```

Comment:
In general, I'm not happy with the responsibility of this class. `ProjectBuilder` and `SolutionBuilder` should not know and care about this issue at all.

Let's move the logic to a higher level (`Verifier`) that is aware of the concurrency issue and pass only the final list of files inside here.